src/electron/npm/libreoffice.d.ts

```diff
@@ -91,49 +92,81 @@ declare namespace LibreOffice {
     redline_table_entry_modified: any;
     redline_table_size_changed: any;
     invalidate_tiles: any;
+  }
+
+  type DocumentEventHandler<
+    Events extends DocumentEvents = DocumentEvents,
+    Event extends keyof Events = keyof Events
+  > = (arg: Events[Event]) => void;
+
+  export type NumberString<T extends number = number> = `${T}`;
+
+  export type UnoString = { type: 'string'; value: string };
+  export type UnoBoolean = { type: 'boolean'; value: boolean };
+  export type UnoLong = { type: 'long'; value: number };
+  export type UnoFloat = { type: 'float'; value: NumberString };
+  export type UnoStringifiedNumber = { type: 'string'; value: NumberString };
+
+  interface UnoCommands {
+    '.uno:SetPageColor': { ColorHex: UnoString };
+    '.uno:FontColor': { FontColor: UnoLong };
+  }
+
+  type CommandValueResult<R = { [name: string]: any }> = {
+    commandValues: R;
   };
 
-  type DocumentEventHandler<Event extends keyof DocumentEvents> = (
-    arg: DocumentEvents[Event]
-  ) => void;
+  interface GetCommands {
+    '.uno:PageColor': string;
```

Comment:
I think we also need `.uno:GetOutline` here

Reply:
The point of this PR is that the actually exposed interfaces are generic, that's effectively a placeholder.

Reply:
Like this:
``` ts
import type { UNO, UnoCommands } from "@coparse/editor-uno-interface";

export interface DocumentEvents extends LibreOffice.DocumentEvents {}
export type DocumentClient = LibreOffice.DocumentClient<DocumentEvents, UNO, UnoCommands>
```

Reply:
Oh ok. Sorry I misunderstood with that being there. Cool beans I'll approve! Looks like lots of fancy typescripting :D